File: create2Dgaussian.m
% Creates a 2D gaussian with dimensions [m,n] 
% Input parameters:
% sigma : standars deviation
% c: position of peak center
% m: width of function
% n: length of function
% Susanna Hammarberg 20170129
function [gauss2D] = create2Dgaussian( sigma, c, width, length)
% allocate memory for matrix
gauss2D = zeros(width,length);
% create a 2DGaussian
for m=1:width
    for n=1:length
        gauss2D(m,n) = exp(-(((n-c).^2)+(m-c).^2)./(2.*sigma.^2));
    end
end

end

File: create2Dgaussian.py
# -*- coding: utf-8 -*-
"""
Created on Tue Mar  7 10:54:48 2017

@author: Susanna Hammarberg
"""
import numpy as np

def create2Dgaussian(sigma, center, width, length):
    gauss2D = np.zeros(shape=(width,length))
    for m in range(0,width): 
       for n in range(0,length):          
            gauss2D[m,n] = np.exp(-(((n-center)**2)+(m-center)**2)/(2*sigma**2))
            

    return gauss2D

    
#x= create2Dgaussian(1,1,1,1)